RocketChat/Rocket.Chat.ReactNative
# feat: capability to enable/disable writing in rooms read only

## Proposed changes


I was working on this task that seemed straightforward at first: relabeling buttons. However, during the tests, we discovered that the 'unmuted' array handling for read-only rooms wasn't functioning correctly on mobile. Surprisingly, the 'unmuted' array wasn't even being saved in the local database. To fix this, we needed more than just label changes. We had to modify the database to handle the 'unmuted' array in the local database.

## Issue(s)	


## How to test or reproduce


- Create a read-only channel and add one or a few users
- unmute users  and try to send a message using them
- then mute again and these users wouldn't be able to send a message again.
- Also check the labels if they are the same as expected by the design

## Screenshots

- At Desktop app, the user logged is the Reinaldo Neto. While on mobile is reinaldo123

https://github.com/RocketChat/Rocket.Chat.ReactNative/assets/47038980/3bfab004-665b-4568-9aa2-3a72aa28fbdf


https://github.com/RocketChat/Rocket.Chat.ReactNative/assets/47038980/30359fa5-de8c-44b7-b8b6-e986efe9d19b


## Types of changes




## Checklist



## Further comments

[TC-48]

[TC-48]: https://rocketchat.atlassian.net/browse/TC-48?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
